### PR TITLE
[Merged by Bors] - Use separate argument to specify dev/prod in smartmodule-publish workflow

### PR DESCRIPTION
**Pull request by ozgrakkurt**

### CODE REVIEW

**Review by sehz — CHANGES_REQUESTED** [review]
Since this is dependent by other repo, we should create another version rather than replace it

**Comment by sehz** [comment]
sounds good

**Review by sehz — APPROVED** [review]

File: .github/workflows/smartmodule-publish.yml [hunk 1]
```diff
@@ -44,7 +40,10 @@ jobs:
       - name: Install Fluvio SMDK
         run: fluvio install smdk
       - name: Fluvio Login
-        run: fluvio cloud login --email ${{ secrets.CLOUD_USER_EMAIL }} --password ${{ secrets.CLOUD_USER_PASSWORD }} ${{ inputs.cloud-url != '' && '--remote' || '' }} ${{ inputs.cloud-url }}
+        run: |
+          fluvio cloud login --email ${{ inputs.target_prod && secrets.HUB_USER_EMAIL || secrets.DEV_HUB_USER_EMAIL }} \
+          --password ${{ inputs.target_prod && secrets.HUB_USER_PASSWORD || secrets.DEV_HUB_USER_PASSWORD }} \
+          ${{ !inputs.target_prod && ('--remote ' + vars.DEV_CLOUD_URL) || '' }}
```
Comment (EstebanBorai) [inline .github/workflows/smartmodule-publish.yml:46]: I think `secrets` must be introduced as input given that these are parameters.
Perhaps its not necessary to put them behind the `secrets` object,

Take a look at how GitHub does here: https://github.com/actions/add-to-project/blob/main/action.yml#L10-L12

The `github-token` input is just another string. I think we could use two regular inputs for User Email and Password.

Comment (sehz) [reply .github/workflows/smartmodule-publish.yml:46]: see: https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository.   GITHUB_TOKEN is special...

Comment (sehz) [reply .github/workflows/smartmodule-publish.yml:46]: and https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow.

**Review by sehz — APPROVED** [review]

**Review by EstebanBorai — APPROVED** [review]

**Pull request closed without merging**